File: algorithms/golang/557.反转字符串中的单词-iii.go

import (
	"strings"
)

/*
 * @lc app=leetcode.cn id=557 lang=golang
 *
 * [557] 反转字符串中的单词 III
 */

// @lc code=start
// reverse will reverse strinrg
func reverse(s string) string {
	size := len(s)
	ans := []rune(s)
	for i := 0; i < size/2; i++ {
		ans[i], ans[size-i-1] = ans[size-i-1], ans[i]
	}
	return string(ans)
}

func reverseWords(s string) string {
	strs := strings.Split(s, " ")
	for i, str := range strs {
		strs[i] = reverse(str)
	}
	return strings.Join(strs, " ")
}

// @lc code=end

File: algorithms/js/557.反转字符串中的单词-iii.js
/*
 * @lc app=leetcode.cn id=557 lang=javascript
 *
 * [557] 反转字符串中的单词 III
 */

// @lc code=start
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
  const ret = [];
  for (let i = 0; i < s.length; i) {
    let start = i;
    while (i < s.length && s[i] != " ") i++;
    for (let k = i - 1; k >= start; k--) {
      ret.push(s[k]);
    }
    while (i < s.length && s[i] == " ") {
      ret.push(" ");
      i++;
    }
  }
  return ret.join("");
};
// @lc code=end
